scripts/hal_alignment/hal_to_maf.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Extract a MAF alignment from a HAL file.
+
+Examples::
+    # Extract a MAF alignment with chromosome 19 of GRCh38 as reference.
+    python hal_to_maf.py --ref-genome GRCh38 --ref-sequence chr19 \
+      --keep-genomes-file genome_list.txt input.hal output.maf
+
+"""
+
+from argparse import ArgumentParser
+import subprocess
+
+from Bio.AlignIO.MafIO import MafIterator, MafWriter
+
+
+if __name__ == '__main__':
+
+    parser = ArgumentParser(description=__doc__)
```

Comment:
Nice trick!

Reply:
In fact, it's not as nice as I'd hoped, because by default the `ArgumentParser` line-wraps the description with less-than-ideal results:
```
usage: hal_to_maf.py [-h] --ref-genome STR --ref-sequence STR --keep-genomes-file PATH
                     [--min-block-seqs INT] [--min-block-size INT]
                     hal_file maf_file

Extract a MAF alignment from a HAL file. Examples:: # Extract a MAF alignment with chromosome 19 of GRCh38
as reference. python hal_to_maf.py --ref-genome GRCh38 --ref-sequence chr19 --keep-genomes-file
genome_list.txt input.hal output.maf

positional arguments:
```

One way to address this is to pass formatter class `RawDescriptionHelpFormatter`, which preserves the description formatting, though this still presents the docstring as plain text:
```
usage: hal_to_maf.py [-h] --ref-genome STR --ref-sequence STR --keep-genomes-file PATH
                     [--min-block-seqs INT] [--min-block-size INT]
                     hal_file maf_file

Extract a MAF alignment from a HAL file.

Examples::
    # Extract a MAF alignment with chromosome 19 of GRCh38 as reference.
    python hal_to_maf.py --ref-genome GRCh38 --ref-sequence chr19 \
      --keep-genomes-file genome_list.txt input.hal output.maf

positional arguments:
```

Another approach would be to keep only the first line of the docstring (`description=__doc__.partition('\n')[0]`), as this is often plain text anyway:
```
usage: hal_to_maf.py [-h] --ref-genome STR --ref-sequence STR --keep-genomes-file PATH
                     [--min-block-seqs INT] [--min-block-size INT]
                     hal_file maf_file

Extract a MAF alignment from a HAL file.

positional arguments:
```
Or I could just buck up and write a description.

Reply:
I would just copy-paste the first line of the docstring as a description :man_shrugging:

Reply:
😆 Go ahead with the most convenient option for you 🙂

Reply:
Done. 🙂